app/src/app/views/stix/matrix/matrix-view/matrix-view.component.ts

```diff
@@ -2,34 +2,44 @@ import { Component, OnInit } from '@angular/core';
 import { Matrix } from 'src/app/classes/stix/matrix';
 import { StixViewPage } from '../../stix-view-page';
 import { RestApiConnectorService } from 'src/app/services/connectors/rest-api/rest-api-connector.service';
-import { StixObject } from 'src/app/classes/stix/stix-object';
 import { AuthenticationService } from 'src/app/services/connectors/authentication/authentication.service';
+import { Tactic } from 'src/app/classes/stix/tactic';
 
 @Component({
     selector: 'app-matrix-view',
     templateUrl: './matrix-view.component.html',
     styleUrls: ['./matrix-view.component.scss']
 })
 export class MatrixViewComponent extends StixViewPage implements OnInit {
-    public all_tactics: Array<StixObject>;
-
+    public all_tactics: Array<Tactic> = [];
+    public view: string = "side";
     public get matrix(): Matrix { return this.config.object as Matrix; }
+    public loaded = false;
 
     constructor(private restAPIConnectorService: RestApiConnectorService, authenticationService: AuthenticationService) {
         super(authenticationService);
     }
 
     ngOnInit() {
+
         if (!this.config.hasOwnProperty('showRelationships') || this.config.showRelationships) {
-            let subscription = this.restAPIConnectorService.getAllTactics().subscribe({
-                next: (all_tactics) => {
-                    this.all_tactics = all_tactics.data;
-                },
+            let subscription = this.restAPIConnectorService.getTechniquesInMatrix(this.matrix).subscribe({
+                next: () => { this.loaded = true
+                this.all_tactics = this.matrix.tactic_objects },
                 complete: () => { subscription.unsubscribe(); } //prevent memory leaks
             })
```

Comment:
The logic for `getAllTactics()` is still needed here, since it is used to build the list of all tactics in the knowledge base that can be added to the matrix. Since the `all_tactics` variable only contains the tactics in the matrix, new tactics cannot be added:

<img width="892" alt="image" src="https://github.com/center-for-threat-informed-defense/attack-workbench-frontend/assets/48832936/24cd91e5-f6c7-4e36-a261-420bdd772478">